com.unity.render-pipelines.universal/Shaders/Utils/StencilDeferred.shader

```diff
@@ -171,14 +171,12 @@ Shader "Hidden/Universal Render Pipeline/StencilDeferred"
 
                 if (!materialReceiveShadowsOff)
                 {
-                    #if defined(MAIN_LIGHT_CALCULATE_SHADOWS)
-                        #if defined(_MAIN_LIGHT_SHADOWS_SCREEN) && !defined(_SURFACE_TYPE_TRANSPARENT)
-                            float4 shadowCoord = float4(screen_uv, 0.0, 1.0);
-                        #else
-                            float4 shadowCoord = TransformWorldToShadowCoord(posWS.xyz);
-                        #endif
-                        unityLight.shadowAttenuation = MainLightShadow(shadowCoord, posWS.xyz, shadowMask, _MainLightOcclusionProbes);
+                    #if defined(_MAIN_LIGHT_SHADOWS_SCREEN) && !defined(_SURFACE_TYPE_TRANSPARENT)
+                        float4 shadowCoord = float4(screen_uv, 0.0, 1.0);
+                    #else
```

Comment:
Should this be?
#elif defined(MAIN_LIGHT_CALCULATE_SHADOWS)
    inputData.shadowCoord = TransformWorldToShadowCoord(inputData.positionWS);
#else
    inputData.shadowCoord = float4(0, 0, 0, 0);
#endif

Does deferred not require the vertex coord interpolator?
#elif defined(REQUIRES_VERTEX_SHADOW_COORD_INTERPOLATOR)
    inputData.shadowCoord = input.shadowCoord;

Reply:
This is the shader that calculates & apply deferred lighting, so there is no vertex coord interpolator passed in from the mesh geometry.

The bug was that `MainLightShadow()` calculates and mixes realtime and baked shadows, so it must always be called.
In the actual FB case, the geometry gets far enough so that `MAIN_LIGHT_CALCULATE_SHADOWS` is not defined anymore, so `MainLightShadow()` wouldn't get called either to apply the baked shadows.

Note that internally `MainLightShadow()` ignores `shadowCoord` if `MAIN_LIGHT_CALCULATE_SHADOWS` is not defined, so  `shadowCoord` computation is compiled out when `MAIN_LIGHT_CALCULATE_SHADOWS` is not defined.

I will make it more explicit.